### PR TITLE
Update go-license-tool

### PR DESCRIPTION
**Pull request by zhenghuiwang**
Update go-license-tool with more detailed instructions and one new script.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/testing/553)

### CODE REVIEW

**Comment by jlewi** [comment]
@Bobgy could you review this please?

File: py/kubeflow/testing/go-license-tools/patch_additional_license_info.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      assert len(parts) == 2
+      [license_url, license_type] = parts
+      mapping[license_url] = license_type
+
+  newlines = []
+  with open(args.license_info_file, 'r') as f:
+    for line in f:
+      parts = line.strip().split(',')
+      _, license_url, license_type, *_ = parts
+      if license_url in mapping and license_type == 'Other':
```
Comment (Bobgy) [inline py/kubeflow/testing/go-license-tools/patch_additional_license_info.py]: For the purpose of the CI workflow, I think this script should better throw an error if some library has license_type 'Other', but it is not in the mapping.
This guards against incomplete patch file.

File: py/kubeflow/testing/go-license-tools/README.md [hunk 2]
```diff
@@ -6,41 +6,137 @@ When we release go library images (can be considered as redistributing third
...
+    ```
+    ......
+    cloud.google.com/go
+    github.com/BurntSushi/toml
+    github.com/beorn7/perks
+    github.com/bmatcuk/doublestar
+    ......
+    ```
 
-    Reminder: don't forget to put the library itself into `dep.txt`.
```
Comment (Bobgy) [inline py/kubeflow/testing/go-license-tools/README.md]: Shall we keep this reminder?
For redistributing 3rd party images, we need the library itself's license.
For distributing first party images, I think it's better we deliver our own license too.

Comment (zhenghuiwang) [reply py/kubeflow/testing/go-license-tools/README.md]: Done. Added to line 43

File: py/kubeflow/testing/go-license-tools/patch_additional_license_info.py (hunk 1 above)
Comment (zhenghuiwang) [reply py/kubeflow/testing/go-license-tools/patch_additional_license_info.py]: Good suggestion! Done.

File: py/kubeflow/testing/go-license-tools/README.md (hunk 2 above)
Comment (zhenghuiwang) [reply py/kubeflow/testing/go-license-tools/README.md]: Done.

**Pull request merged**